needle.pm

```diff
@@ -276,13 +276,34 @@ sub has_tag {
 }
 
 sub has_property {
-    my ($self, $tag) = @_;
+    my ($self, $pro) = @_;
     for my $t (@{$self->{properties}}) {
-        return 1 if ($t eq $tag);
+        if (ref($t) eq "HASH") {
+            return 1 if ($t->{name} eq $pro);
+        }
+        else {
+            return 1 if ($t eq $pro);
+        }
     }
     return 0;
 }
 
+sub get_property_value {
+    my ($self, $pro) = @_;
+    for my $t (@{$self->{properties}}) {
+        if (ref($t) eq "HASH") {
+            return $t->{value} if ($t->{name} eq $pro);
+        }
+        else {
+            return "" if ($pro ne "workaround");
```

Comment:
Wouldn't this function fail to find a property value if a non-hash comes before? So shouldn't it be `next` instead of `return`?